mvp_example/src/main/java/com/distillery/android/blueprints/mvp/adapter/ToDoListAdapter.kt

```diff
@@ -0,0 +1,57 @@
+package com.distillery.android.blueprints.mvp.adapter
+
+import android.graphics.Paint.STRIKE_THRU_TEXT_FLAG
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import com.distillery.android.ui.databinding.ItemTodoBinding
+import com.distillery.android.domain.models.ToDoModel
+
+class ToDoListAdapter(
+    private val onDeleteClickListener: (toDoModel: ToDoModel) -> Unit,
+    private val onCompleteClickListener: (toDoModel: ToDoModel) -> Boolean
+) :
+        ListAdapter<ToDoModel, ToDoListAdapter.ToDoViewHolder>(ToDoListDiffCallback()) {
+
+    class ToDoViewHolder(private val binding: ItemTodoBinding) : RecyclerView.ViewHolder(binding.root) {
+        fun bind(
+            toDoModel: ToDoModel,
+            onDeleteClickListener: (toDoModel: ToDoModel) -> Unit,
+            onCompleteClickListener: (toDoModel: ToDoModel) -> Boolean
+        ) {
+            binding.titleTextView.text = toDoModel.title
+            binding.descriptionTextView.text = toDoModel.description
+            binding.deleteButton.setOnClickListener {
+                onDeleteClickListener(toDoModel)
+            }
```

Comment:
`with` or `.apply` can be used to avoid repeating `binding.` all over the method, e.g.
```suggestion
            binding.apply {
                titleTextView.text = toDoModel.title
                descriptionTextView.text = toDoModel.description
                deleteButton.setOnClickListener {
                    onDeleteClickListener(toDoModel)
                }
```